src/components/Shared/GeneralLedger.js

```diff
@@ -19,193 +20,378 @@ import CustomDatePicker from 'src/components/Inputs/CustomDatePicker'
+
+        const data = {
+          transactions: values.generalAccount.map(({ id, exRate, tpAccount, functionId, ...rest }) => ({
+            seqNo: id,
+
+            exRate,
+
+            functionId,
+            rateCalcMethod: 1,
+
+            ...rest
+          })),
+          date: formatDateToApi(values.date),
+          functionId: values.functionId,
+          recordId: formValues.recordId,
+          reference: values.reference
         }
-      },{
-        field:"notes",
-        headerName:_labels.notes,
-        flex:1,
-      },{
-        field:"exRate",
-        headerName:_labels.exRate,
-        flex:1,
-        align: 'right',
-      },{
-        field:"amount",
-        headerName:_labels.amount,
-        flex:1,
-        align: 'right',
-        
-      },{
-        field:"baseAmount",
-        headerName:_labels.base,
-        flex:1,
-        align: 'right',
-  
+
+        console.log('Submitting data:', data)
+
+        const response = await postRequest({
+          extension: GeneralLedgerRepository.GeneralLedger.set2,
+          record: JSON.stringify(data)
+        })
+
+        console.log('Submission response:', response)
+
+        toast.success('Record Added Successfully')
       }
+    }
+  })
 
-    ]
+  useEffect(() => {
+    if (formValues) {
+      setformik(formValues)
+    }
+  }, [formValues])
 
-    useEffect(() => {
-      if (data && data.list && Array.isArray(data.list)) {
-        const baseCredit = data.list.reduce((acc, curr) => curr.sign === 2 ? acc + parseFloat(curr.baseAmount || 0) : acc, 0);
-        const baseDebit = data.list.reduce((acc, curr) => curr.sign === 1 ? acc + parseFloat(curr.baseAmount || 0) : acc, 0);
-        const baseBalance = baseDebit - baseCredit;
-        
-        setBaseGridData({ base :'Base',credit: baseCredit, debit: baseDebit, balance: baseBalance });
-    
-        const currencyTotals = data.list.reduce((acc, curr) => {
-          const currency = curr.currencyRef;
-          if (!acc[currency]) {
-            acc[currency] = { credit: 0, debit: 0 };
-          }
-          if (curr.sign === 2) {
-            acc[currency].credit += parseFloat(curr.amount || 0);
-          } else if (curr.sign === 1) {
-            acc[currency].debit += parseFloat(curr.amount || 0);
+  const {
+    query: { data },
+    labels: _labels,
+    access
+  } = useResourceQuery({
+    queryFn: fetchGridData,
+
+    datasetId: ResourceIds.GeneralLedger
+  })
+
+  useEffect(() => {
+    if (formik2 && formik2.values && formik2.values.generalAccount && Array.isArray(formik2.values.generalAccount)) {
+      const generalAccountData = formik2.values.generalAccount
+
+      console.log(generalAccountData)
+
+      const parseNumber = value => {
+        const number = parseFloat(value)
+
+        return isNaN(number) ? 0 : number
+      }
+
+      const baseCredit = generalAccountData.reduce((acc, curr) => {
+        return curr.signName === 'C' ? acc + parseNumber(curr.baseAmount) : acc
```

Comment:
never compare to the name, always use the key or the ID. here you should use curr.sign 
you should take into consideration that Argus is multiling, in arabic they won't return C and D as signName.